### DISCUSSION TITLE
Where is the console listened to at?

### DISCUSSION THREAD

**Discussion by amnoah** [Minestom/Minestom#1484] [Help] [answered]
Looking into how some of the functions of Minestom work to implement in a personal project.

Where exactly does Minestom listen to messages sent in console?

Example: If I type "Hello" in the console window, Minestom will attempt to run "Hello" as a command due to listening to the console.

**Comment by TheMode** [comment, marked as answer]
In the [`terminal`](https://github.com/Minestom/Minestom/tree/master/src/main/java/net/minestom/server/terminal) package:

**Reply by amnoah** [reply]
Thank you!
